### ISSUE TITLE
cloudflare_tunnel_config ingress rules have extra origin_request configuration

### ISSUE THREAD

**Issue by ahrenstein** [cloudflare/terraform-provider-cloudflare#2520]
### Confirmation

- [X] My issue isn't already found on the issue tracker.
- [X] I have replicated my issue using the latest version of the provider and it is still present.

### Terraform and Cloudflare provider version

Terraform v1.5.0
on darwin_arm64
+ provider registry.terraform.io/cloudflare/cloudflare v4.7.1
+ provider registry.terraform.io/hashicorp/aws v4.9.0
+ provider registry.terraform.io/hashicorp/random v3.5.1
+ provider registry.terraform.io/hashicorp/tls v3.3.0

### Affected resource(s)

1. cloudflare_tunnel_config

### Terraform configuration files

```hcl
resource "cloudflare_tunnel" "tunnel" {
  account_id = var.cloudflare_account_id
  name       = "leet-tunnel"
  secret     = base64encode("[REDACTED]")
}

resource "cloudflare_tunnel_config" "tunnel" {
  account_id = var.cloudflare_account_id
  tunnel_id  = cloudflare_tunnel.tunnel.id
  config {
    ingress_rule {
      hostname = "test-tunnel.example.com"
      service  = "https://nginx.webservers.svc.cluster.local:443"
      origin_request = {
        no_tls_verify = true
      }
    }
    ingress_rule {
      service = "http_status:404"
    }
  }
}
```

### Link to debug output

https://gist.github.com/ahrenstein/1f12a5a20ff166a1417c8b0a50041e71

### Expected output

```hcl
+ ingress_rule {
              + hostname = "test-tunnel.example.com"
              + service  = "https://nginx.webservers.svc.cluster.local:443"

              + origin_request {
                  + no_tls_verify            = true
                }
            }
```

### Actual output

```hcl
+ ingress_rule {
              + hostname = "test-tunnel.example.com"
              + service  = "https://nginx.webservers.svc.cluster.local:443"

              + origin_request {
                  + connect_timeout          = "30s"
                  + disable_chunked_encoding = false
                  + http2_origin             = false
                  + keep_alive_connections   = 100
                  + keep_alive_timeout       = "1m30s"
                  + no_happy_eyeballs        = false
                  + no_tls_verify            = true
                  + proxy_address            = "127.0.0.1"
                  + proxy_port               = 0
                  + tcp_keep_alive           = "30s"
                  + tls_timeout              = "10s"
                }
            }
```

### Steps to reproduce

1. Create a terraformed `cloudflare_tunnel` resource and respective `cloudflare_tunnel_config` resource.
    1. The tunnel configuration should have an ingress rule where `origin_request` only contains a single item such as `no_tls_verify`
2. Run `terraform apply`

Expected: Output should only adjust origin request settings on Cloudflare for `no_tls_verify`
Actual: 7 total adjustments are made, including the expected one.

### Additional factoids

Setting additional `origin_request` rules other than the specified one(s) results in extra configuration above the defaults you would find via hand creating the ingress rules in the Zero Trust console directly. This is not proper as defaults should be left in place unless specified.

**Comment by ahrenstein** [comment]
https://gist.github.com/ahrenstein/1f12a5a20ff166a1417c8b0a50041e71

**Comment by jacobbednarz** [comment]
addressed by #2504

**Comment by mmalyska** [comment]
@jacobbednarz but will it fix also problems in `ingress_rule.origin_request`? It also adds `proxy_address` and other props that are not a duration, but should be empty.

Adding `origin_request` in new plan after creating `ingress_rules`
```
esource "cloudflare_tunnel_config" "jaskinia_config" {
  account_id = cloudflare_account.main.id
  tunnel_id  = cloudflare_tunnel.jaskinia.id

  config {
    ingress_rule {
      hostname = "${local.cloudflare_domain}"
      service  = "https://traefik.traefik.svc.cluster.local:443"
      origin_request {
        origin_server_name = cloudflare_record.ingress.hostname
      }
    }
    ingress_rule {
      hostname = "*.${local.cloudflare_domain}"
      service  = "https://traefik.traefik.svc.cluster.local:443"
      origin_request {
        origin_server_name = cloudflare_record.ingress.hostname
      }
    }
    ingress_rule {
      service = "http_status:404"
    }
  }
}
```

generates plan:
```
# cloudflare_tunnel_config.jaskinia_config will be updated in-place
  ~ resource "cloudflare_tunnel_config" "jaskinia_config" {
        id         = "{some id}"
        # (2 unchanged attributes hidden)

      ~ config {
          ~ ingress_rule {
                # (2 unchanged attributes hidden)

              + origin_request {
                  + connect_timeout          = "30s"
                  + disable_chunked_encoding = false
                  + http2_origin             = false
                  + keep_alive_connections   = 100
                  + keep_alive_timeout       = "1m30s"
                  + no_happy_eyeballs        = false
                  + no_tls_verify            = false
                  + origin_server_name       = "ingress.{domain}"
                  + proxy_address            = "127.0.0.1"
                  + proxy_port               = 0
                  + tcp_keep_alive           = "30s"
                  + tls_timeout              = "10s"
                }
            }
          ~ ingress_rule {
                # (2 unchanged attributes hidden)

              + origin_request {
                  + connect_timeout          = "30s"
                  + disable_chunked_encoding = false
                  + http2_origin             = false
                  + keep_alive_connections   = 100
                  + keep_alive_timeout       = "1m30s"
                  + no_happy_eyeballs        = false
                  + no_tls_verify            = false
                  + origin_server_name       = "ingress.{domain}"
                  + proxy_address            = "127.0.0.1"
                  + proxy_port               = 0
                  + tcp_keep_alive           = "30s"
                  + tls_timeout              = "10s"
                }
            }

            # (1 unchanged block hidden)
        }
    }
```

It should only add `origin_server_name` and not override any defaults. Also the durations are a bit wonky.
They looks this way in CF UI after applying the plan:
![obraz](https://github.com/cloudflare/terraform-provider-cloudflare/assets/999598/963d7750-67ed-4600-a1a7-2741a5c4197d)
and they are not correct as durations base value is seconds and the values are too big:
![obraz](https://github.com/cloudflare/terraform-provider-cloudflare/assets/999598/c5d790bd-47d4-4367-baf0-cab66dca4941)

After fixing them in UI and running plan it treats them as `ns` and not `s`:
```
# cloudflare_tunnel_config.jaskinia_config will be updated in-place
  ~ resource "cloudflare_tunnel_config" "jaskinia_config" {
        id         = "{some id}"
        # (2 unchanged attributes hidden)

      ~ config {
          ~ ingress_rule {
                # (2 unchanged attributes hidden)

              ~ origin_request {
                  ~ connect_timeout          = "30ns" -> "30s"
                  ~ keep_alive_timeout       = "90ns" -> "1m30s"
                  ~ tcp_keep_alive           = "30ns" -> "30s"
                  ~ tls_timeout              = "10ns" -> "10s"
                    # (9 unchanged attributes hidden)
                }
            }
          ~ ingress_rule {
                # (2 unchanged attributes hidden)

              ~ origin_request {
                  ~ connect_timeout          = "30ns" -> "30s"
                  ~ keep_alive_timeout       = "90ns" -> "1m30s"
                  ~ tcp_keep_alive           = "30ns" -> "30s"
                  ~ tls_timeout              = "10ns" -> "10s"
                    # (9 unchanged attributes hidden)
                }
            }

            # (1 unchanged block hidden)
        }
    }
```

So I think this shouldn't be closed as it won't fix adding additional properties in `ingress_rule.origin_request`.
